core/src/main/scala/sbtorgpolicies/github/GitHubOps.scala

```diff
@@ -50,7 +54,30 @@ class GitHubOps(owner: String, repo: String, accessToken: Option[String]) {
       repo: String,
       branch: String,
       message: String,
-      filesAndContents: List[(String, String)]): Either[GitHubException, Ref] = {
+      files: List[String]): Either[OrgPolicyException, Ref] = {
+
+    def readFileContents: IOResult[List[(String, String)]] = {
+      files.foldLeft[IOResult[List[(String, String)]]](Right(Nil)) {
+        case (Right(partialResult), file) =>
+          fileReader.getFileContent(file).map(partialResult :+ (file, _))
+        case (Left(e), _) => Left(e)
+      }
+    }
+
+    readFileContents match {
```

Comment:
Not possible because we're calling to a function that returns an `Either[GithubException, T]`